pkg/apm/components.go

```diff
@@ -0,0 +1,228 @@
+package apm
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// ListComponentsParams represents a set of filters to be
+// used when querying New Relic applications.
+type ListComponentsParams struct {
+	Name         string
+	IDs          []int
+	PluginID     int
+	HealthStatus bool
+}
+
+// ListComponents is used to retrieve the components associated with
+// a New Relic account.
+func (apm *APM) ListComponents(params *ListComponentsParams) ([]Component, error) {
+	response := componentsResponse{}
+	c := []Component{}
+	nextURL := "/components.json"
+	paramsMap := buildListComponentsParamsMap(params)
+
+	for nextURL != "" {
+		resp, err := apm.client.Get(nextURL, &paramsMap, &response)
+
+		if err != nil {
+			return nil, err
+		}
+
+		c = append(c, response.Components...)
+
+		paging := apm.pager.Parse(resp)
+		nextURL = paging.Next
+	}
+
+	return c, nil
+
+}
+
+// GetComponent is used to retrieve a specific New Relic component.
+func (apm *APM) GetComponent(componentID int) (*Component, error) {
+	response := componentResponse{}
+	url := fmt.Sprintf("/components/%d.json", componentID)
+
+	_, err := apm.client.Get(url, nil, &response)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &response.Component, nil
+}
+
+// ListComponentMetricsParams represents a set of parameters to be
+// used when querying New Relic component metrics.
+type ListComponentMetricsParams struct {
+	// Name allows for filtering the returned list of metrics by name.
+	Name string
+}
+
+// ListComponentMetrics is used to retrieve the metrics for a specific New Relic component.
+func (apm *APM) ListComponentMetrics(componentID int, params *ListComponentMetricsParams) ([]ComponentMetric, error) {
```

Comment:
Same question as above regarding returning array of pointers 😎

Reply:
👍